common/cache/lru.go

```diff
@@ -169,6 +171,10 @@ func (c *lru) Get(key interface{}) interface{} {
 	c.mut.Lock()
 	defer c.mut.Unlock()
 
+	if c.activelyEvict {
```

Comment:
Instead of spreading the condition around, I suggest to call a function that checks this in one place.
```suggestion
		c.processEvicted()
```

And inside 
```
func (c *lru) processEvicted() {
   if c.activelyEvict {
      return
   }
```